include/swift/AST/Attr.h

```diff
@@ -1802,6 +1902,11 @@ class CustomAttr final : public DeclAttribute {
     return DA->getKind() == DeclAttrKind::Custom;
   }
 
+  /// Create a copy of this attribute.
+  CustomAttr *clone(ASTContext &ctx) const {
+    return create(ctx, AtLoc, getTypeExpr(), isImplicit());
```

Comment:
`ArgumentList *argList;` and `PatternBindingInitializer *initContext;` are not cloned. Do we need to support `CustomAttr` in swift_attrs? If not, we should make this `UNIMPLEMENTED_CLONE`, then skip `CustomAttr` before cloning in ImportDecl (and diagnose).

Reply:
We do need `CustomAttr`, yes.